ObservatoryCore/ObservatoryCore.cs

```diff
@@ -10,13 +10,39 @@ class ObservatoryCore
         static void Main(string[] args)
         {
             string version = System.Reflection.Assembly.GetEntryAssembly().GetName().Version.ToString();
-            if (Properties.Core.Default.CoreVersion != version)
+            try
             {
-                Properties.Core.Default.Upgrade();
-                Properties.Core.Default.CoreVersion = version;
-                Properties.Core.Default.Save();
+                if (Properties.Core.Default.CoreVersion != version)
+                {
+                    Properties.Core.Default.Upgrade();
+                    Properties.Core.Default.CoreVersion = version;
+                    Properties.Core.Default.Save();
+                }
+                BuildAvaloniaApp().StartWithClassicDesktopLifetime(args);
+                throw new Exception("Test Exception", new Exception("Inner Exception Test"));
             }
-            BuildAvaloniaApp().StartWithClassicDesktopLifetime(args);
+            catch (Exception ex)
+            {
+                var docPath = System.Environment.GetFolderPath(System.Environment.SpecialFolder.MyDocuments);
+                var errorMessage = new System.Text.StringBuilder();
+                errorMessage.AppendLine($"Error encountered in Elite Observatory {version}.");
```

Comment:
Optional: Consider chaining:

errorMessage
    .AppendLine(...)
    .AppendLine(...)
    .AppendLine();

(Of course, however the editor formats it. Below as well.)